infra/modules/azure_role_assignments/variables.tf

```diff
@@ -0,0 +1,87 @@
+variable "principal_id" {
+  description = "The ID of the principal to which assign roles. It can be a managed identity."
+  type        = string
+}
+
+variable "cosmos" {
+  description = "A list of CosmosDB role assignments"
+  type = list(object({
+    account_name        = string
+    resource_group_name = string
+    role                = string
```

Comment:
Is this value supposed to be validated to consider only `writer` and `reader` as valid values?

Reply:
Right! Validation will be put in place asap